### PR TITLE
Merge 3.0 => 3.1

### PR DESCRIPTION
**Pull request by crummel**

### CODE REVIEW

File: support/allowed-prebuilts.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (dseefeld) [inline support/allowed-prebuilts.txt]: There are still some roslyn prebuilts in 3.1.

File: support/additional-prebuilts-to-delete.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# this is a list of packages that end up in the tarball's prebuilts folder but are not actually needed.
```
Comment (dseefeld) [inline support/additional-prebuilts-to-delete.txt]: Will this list need to be updated for 3.1?

File: support/allowed-prebuilts.txt (hunk 1 above)
Comment (crummel) [reply support/allowed-prebuilts.txt]: I disabled actually prebuilt enforcement with https://github.com/dotnet/source-build/pull/1383/commits/dc9a11ced5fd7fbf8bed5d8d9b460f0c708ed54c, I figured once we hit zero it should be easy to flip that back on and with your improvement we'll only need to run CI once to find all the extras.

File: support/additional-prebuilts-to-delete.txt (hunk 2 above)
Comment (crummel) [reply support/additional-prebuilts-to-delete.txt]: Yes, but I was going to wait until we hit zero in the baseline, I think that will be easier.

File: support/allowed-prebuilts.txt (hunk 1 above)
Comment (dseefeld) [reply support/allowed-prebuilts.txt]: Got it.  Makes sense.

**Comment by crummel** [comment]
CentOS Production passed on re-run but AzDo isn't updating the status (https://dev.azure.com/dnceng/public/_build/results?buildId=434521&view=logs&jobId=935c00b8-5de8-597d-a783-42e7f079746c).  Merging.

**Pull request merged**